test/DynamoCoreTests/SerializationTests.cs

```diff
@@ -274,53 +236,72 @@ private void CompareWorkspacesDifferentGuids(WorkspaceComparisonData a, Workspac
             if (nodeDiff.Any())
             {
-                Assert.Fail("The workspaces don't have the same number of nodes. The json workspace is missing: " + string.Join(",", nodeDiff.Select(i => i.Value.ToString())));
+                Assert.Fail("The workspaces don't have the same number of nodes. The json workspace is missing: " + string.Join(",", nodeDiff.Select(i => i.ToString())));
             }
-            Assert.AreEqual(a.Description, b.Description, "The workspaces don't have the same description.");
             Assert.AreEqual(a.NodeCount, b.NodeCount, "The workspaces don't have the same number of nodes.");
             Assert.AreEqual(a.ConnectorCount, b.ConnectorCount, "The workspaces don't have the same number of connectors.");
-            //TODO: Annotations / Note tests should be in viewmodel serialization tests.
-            // Assert.AreEqual(a.GroupCount, b.GroupCount, "The workspaces don't have the same number of groups.");
-            // Assert.AreEqual(a.NoteCount, b.NoteCount, "The workspaces don't have the same number of notes.");
+
             foreach (var kvp in a.InportCountMap)
             {
                 var countA = kvp.Value;
-                var countB = b.InportCountMap[kvp.Key];
+                //convert the old guid to the new guid
+                var newGuid = GuidUtility.Create(GuidUtility.UrlNamespace, modelGuidsToIDmap[kvp.Key]);
+                var countB = b.InportCountMap[newGuid];
                 Assert.AreEqual(countA, countB, string.Format("One {0} node has {1} inports, while the other has {2}", a.NodeTypeMap[kvp.Key], countA, countB));
             }
             foreach (var kvp in a.OutportCountMap)
             {
                 var countA = kvp.Value;
-                var countB = b.OutportCountMap[kvp.Key];
+                //convert the old guid to the new guid
+                var newGuid = GuidUtility.Create(GuidUtility.UrlNamespace, modelGuidsToIDmap[kvp.Key]);
+                var countB = b.OutportCountMap[newGuid];
                 Assert.AreEqual(countA, countB, string.Format("One {0} node has {1} outports, while the other has {2}", a.NodeTypeMap[kvp.Key], countA, countB));
             }
 
             foreach (var portkvp in a.PortDataMap)
             {
-                Assert.IsTrue(b.PortDataMap.ContainsKey(portkvp.Key));
-                Assert.AreEqual(a.PortDataMap[portkvp.Key], b.PortDataMap[portkvp.Key]);
+                //convert the old guid to the new guid
+                var newGuid = GuidUtility.Create(GuidUtility.UrlNamespace, modelGuidsToIDmap[portkvp.Key]);
+                Assert.IsTrue(b.PortDataMap.ContainsKey(newGuid));
+                var aPort = a.PortDataMap[portkvp.Key];
+                var bPort = b.PortDataMap[newGuid];
+                Assert.AreEqual(aPort.UseLevels, bPort.UseLevels);
+                Assert.AreEqual(aPort.KeepListStructure, bPort.KeepListStructure);
+                Assert.AreEqual(aPort.Level, bPort.Level);
```

Comment:
no description comparison here?

Reply:
will add it, not sure why it's not there.